database.go

```diff
@@ -0,0 +1,26 @@
+package eremetic
+
+import "encoding/json"
+
+// TaskDB defines the functions needed by the database abstraction layer
+type TaskDB interface {
+	Clean() error
+	Close()
+	PutTask(task *Task) error
+	ReadTask(id string) (Task, error)
+	ReadUnmaskedTask(id string) (Task, error)
+	ListNonTerminalTasks() ([]*Task, error)
+}
+
+const Masking = "*******"
+
+func ApplyMask(task *Task) {
```

Comment:
I agree, this PR (and the previous one) are really organizational changes and I've gone with the path of the least resistance here. I believe it will enable further refactoring work like the one you mentioned.